addons/mail/models/mail_followers.py

```diff
@@ -142,24 +142,23 @@ def _get_recipient_data(self, records, message_type, subtype_id, pids=None, cids
         LEFT JOIN res_users users ON users.partner_id = partner.id AND users.active
         LEFT JOIN res_groups_users_rel groups_rel ON groups_rel.uid = users.id
         LEFT JOIN res_groups groups ON groups.id = groups_rel.gid
-        WHERE EXISTS (
-            SELECT partner_id FROM sub_followers
-            WHERE sub_followers.channel_id IS NULL
-                AND sub_followers.partner_id = partner.id
-                AND (coalesce(sub_followers.internal, false) <> TRUE OR coalesce(partner.partner_share, false) <> TRUE)
-        ) %s
+        LEFT JOIN sub_followers subfol ON subfol.partner_id = partner.id
```

Comment:
Why `LEFT JOIN` instead of `JOIN` ?

Reply:
Is ``JOIN`` equal to ``INNER JOIN``?
It can't be used, because  condition is ``partner.id IN %s OR subfol....``

Reply:
But we can use ``JOIN`` if we split request into two requests

Reply:
My point is `WHERE EXISTS (SELECT partner_id FROM sub_followers WHERE sub_followers.partner_id = partner.id` is equivalent OF `JOIN sub_followers ON sub_followers.partner_id = partner.id` 

LEFT JOIN will retrieve partners without sub_followers, such records will then be discarded by the `WHERE subfol.partner_id = partner.id` because sub_followers.partner_id cannot be null AND equal to a given ID.

Reply:
I don't know why, but it works slower. You can find a link to explain plan in the [task](https://www.odoo.com/web#action=4043&cids=1&id=2411637&menu_id=4720&model=project.task&view_type=form)